### PR TITLE
Create flexible DataFeedInjector system to allow for arbitrary item injection and result replacement

### LINKED ISSUE #6: Allow mods to inject elements in arbitrary positions in the settings
This would allow mods that add things specific to certain subsystems to inject their settings right with the settings for that subsystem, making it easier to configure in one place.

For example: allowing a mod that adds notifications to add their settings to the existing notification settings.

### PR DESCRIPTION
**Pull request by Banane9**
Closes #6 and #9.

### CODE REVIEW

**Comment by Banane9** [comment]
If you want to test this, you also need the latest version of MonkeyLoader - make sure to move the new EnumerableToolkit* dependencies to the Resonite/MonkeyLoader/ folder too

**Comment by Nytra** [comment]
Settings template injection not working

**Comment by Nytra** [comment]
Requires `System.Linq.Async` which isn't included with MonkeyLoader

**Review by Nytra — DISMISSED** [review]
See previous comments

**Comment by Nytra** [comment]
```
Cannot yield to world, origin CoroutineManager is null!

   at void Elements.Core.UniLog.Error(string message, bool stackTrace)
   at void FrooxEngine.NextUpdate.OnCompleted(Action continuation)
   at void System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Threading.Tasks.VoidTaskResult>.AwaitOnCompleted<NextUpdate, <<-ctor>b__25_0>d>(ref NextUpdate awaiter, ref <<-ctor>b__25_0>d stateMachine)
   at void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.AwaitOnCompleted<NextUpdate, <<-ctor>b__25_0>d>(ref NextUpdate awaiter, ref <<-ctor>b__25_0>d stateMachine)
   at async MonkeyLoader.Resonite.DataFeeds.Settings.SettingsViewData(SettingsDataFeed dataFeed)+(?) => { }
   at void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<<<-ctor>b__25_0>d>(ref <<-ctor>b__25_0>d stateMachine)
   at MonkeyLoader.Resonite.DataFeeds.Settings.SettingsViewData(SettingsDataFeed dataFeed)+() => { }
   at Task FrooxEngine.ComponentBase<C>.RunSynchronouslyAsync(Action action, bool immediatellyIfPossible)+() => { }
   at void FrooxEngine.World.RunSynchronousActions()
   at void FrooxEngine.World.NextRefreshStage()
   at bool FrooxEngine.World.Refresh()
   at void FrooxEngine.WorldManager.UpdateStep()
   at bool FrooxEngine.WorldManager.RunUpdateLoop()
   at void FrooxEngine.Engine.UpdateStep()
   at void FrooxEngine.Engine.RunUpdateLoop()
   at void UnityFrooxEngineRunner.FrooxEngineRunner.UpdateFrooxEngine()
   at void UnityFrooxEngineRunner.FrooxEngineRunner.Update()
```

File: MonkeyLoader.Resonite.Integration/DataFeeds/Settings/SettingsViewData.cs [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+                return;
+
+            RootCategoryView = dataFeed.Slot.GetComponent<RootCategoryView>();
+
+            dataFeed.RunSynchronouslyAsync(async () =>
+            {
+                foreach (var primitiveType in GetTemplateTypes())
+                {
+                    await default(NextUpdate);
+                    EnsureDataFeedValueFieldTemplate(primitiveType);
```
Comment (Nytra) [inline MonkeyLoader.Resonite.Integration/DataFeeds/Settings/SettingsViewData.cs:79]: This causes a significant amount of slowdown on startup despite the template generation being spaced out. I think it was better before when the templates were generated on-demand.

Comment (Banane9) [reply MonkeyLoader.Resonite.Integration/DataFeeds/Settings/SettingsViewData.cs:84]: Yea... We can move it back to be on demand too, just need to call that method when making the value fields as it used to be

**Review by Nytra — APPROVED** [review]
Seems good

**Pull request merged**